src/components/Shared/AmountInput/AmountInput.tsx

```diff
@@ -14,33 +13,75 @@ export type Props = {
 
 const AmountInput: FC<Props> = ({ amount, register, errorMessage }) => {
   const { t } = useTranslation();
+  const [amountInput, setAmountInput] = useState<string>(
+    amount ? "" + amount : ""
+  );
+  const { unit, toggleUnit } = useContext(AppContext);
+
+  const { onChange } = register;
 
-  const ButtonToggleUnit: FC = () => {
-    const { unit, toggleUnit } = useContext(AppContext);
+  const toggleHandler = () => {
+    let formattedValue = amountInput;
+    if (unit === Unit.BTC) {
+      formattedValue = new Intl.NumberFormat("en-US").format(
+        convertBtcToSat(+formattedValue)!
+      );
+      setAmountInput(formattedValue);
+    } else {
+      formattedValue = formattedValue.replace(/,|\./g, "");
+      console.log(+convertSatToBtc(+formattedValue)!);
+      formattedValue = new Intl.NumberFormat("en-US", {
+        minimumFractionDigits: 8,
+      }).format(convertSatToBtc(parseInt(formattedValue))!);
+      setAmountInput(formattedValue);
+    }
+    toggleUnit();
+    onChange({ target: { value: formattedValue } });
+  };
+
+  const onChangeHandler = (e: ChangeEvent<HTMLInputElement>) => {
+    let value = e.target.value;
+    value = value.replace(/[^0-9.,]/, "");
+    if (unit === Unit.SAT) {
+      value = value.replace(/,|\./g, "");
```

Comment:
Or if we use these regexs in different places, can those be defiend once and commented and placed in needed places?